dataplug-twitter/app/org/hatdex/dataplugTwitter/apiInterfaces/TwitterFriendInterface.scala

```diff
@@ -39,69 +39,107 @@ class TwitterFriendInterface @Inject() (
-
   val refreshInterval = 24.hours
 
   def buildContinuation(content: JsValue, params: ApiEndpointCall): Option[ApiEndpointCall] = {
-    val maybeUsers = (content \ "users").asOpt[JsArray]
-    val maybeNextCursor = (content \ "next_cursor_str").asOpt[String]
-
-    params.pathParameters.get("userId").map { userId =>
-      val listContainsUser = maybeUsers map { users =>
-        users.value.exists(v => (v \ "id_str").asOpt[String].contains(userId))
-      } getOrElse {
-        false
+    (content \ "next_cursor_str").as[String] match {
+      case "0" =>
+        logger.debug("Cursor is 0")
+        None
+
+      case cursor: String => {
+        logger.debug("Cursor found")
+        val tempParams = checkIfWeHaveNewFollowers(content, checkForMostRecentFollower(content, params))
+        tempParams match {
+          case Some(parameters) =>
+            val maybeMostRecentFollower = params.storageParameters.flatMap(_.get("mostRecentFollower"))
+            logger.debug(s"Most recent follower in continuation has been found and it's: $maybeMostRecentFollower")
+            maybeMostRecentFollower match {
+              case Some(_) => Some(parameters.copy(queryParameters = parameters.queryParameters + ("cursor" -> "-1")))
+              case _       => Some(parameters.copy(queryParameters = parameters.queryParameters + ("cursor" -> cursor)))
+            }
+
+          case _ =>
+            logger.debug(s"No new parameters")
+            None
+        }
       }
+    }
+  }
 
-      val userFirstInList = maybeUsers map { users =>
-        users.value.headOption
-          .map(firstUser => (firstUser \ "id_str").asOpt[String].contains(userId))
-          .isDefined
-      } getOrElse {
-        false
-      }
+  def buildNextSync(content: JsValue, params: ApiEndpointCall): ApiEndpointCall = {
+    updateStorageParametersMostRecentFollower(content, params)
+  }
 
-      val result: Option[ApiEndpointCall] = (listContainsUser, userFirstInList, maybeNextCursor) match {
-        // Stop continuation, nothing to update
-        case (true, true, _)       => None
-        // Update HAT with new users and reset to the latest user ID
-        case (true, false, _)      => Some(updatedParamsWithFirstUserId(maybeUsers, params))
-        // Stop continuation, last page reached
-        case (false, _, Some("0")) => None
-        // Build continuation for the next page
-        case (false, _, _) =>
-          maybeNextCursor map { nextCursor =>
-            val update = params.queryParameters + ("cursor" -> nextCursor)
-            params.copy(queryParameters = update)
-          }
-      }
+  private def checkForMostRecentFollower(content: JsValue, params: ApiEndpointCall): ApiEndpointCall = {
+    val maybeMostRecentFollower = params.storageParameters.flatMap(_.get("tempMostRecentFollower"))
+    maybeMostRecentFollower match {
+      case None =>
+        (content \ "users").as[JsArray].value.head.validate[TwitterUser].map { user =>
```

Comment:
can be shortened by wrapping the search of TwitterUser ID field into Json path lookup in a Try, then handle found/not-found cases